community/cypher/cypher-compiler-3.4/src/test/scala/org/neo4j/cypher/internal/compiler/v3_4/ast/rewriters/TransitiveClosureTest.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compiler.v3_4.ast.rewriters
+
+import org.neo4j.cypher.internal.compiler.v3_4.phases.LogicalPlanState
+import org.neo4j.cypher.internal.compiler.v3_4.planner.AstRewritingTestSupport
+import org.neo4j.cypher.internal.compiler.v3_4.test_helpers.ContextHelper
+import org.neo4j.cypher.internal.frontend.v3_4.ast.Query
+import org.neo4j.cypher.internal.frontend.v3_4.ast.rewriters.{rewriteEqualityToInPredicate, transitiveClosure}
+import org.neo4j.cypher.internal.util.v3_4.test_helpers.CypherFunSuite
+
+class TransitiveClosureTest extends CypherFunSuite with AstRewritingTestSupport {
+
+  test("MATCH (a)-->(b) WHERE a.prop = b.prop AND b.prop = 42") {
+    shouldRewrite(
+      "MATCH (a)-->(b) WHERE a.prop = b.prop AND b.prop = 42",
+      "MATCH (a)-->(b) WHERE a.prop = 42 AND b.prop = 42")
+  }
+
+  test("MATCH (a)-->(b) WHERE b.prop = a.prop AND b.prop = 42") {
+    shouldRewrite(
+      "MATCH (a)-->(b) WHERE a.prop = b.prop AND b.prop = 42",
```

Comment:
Test title differs in equality order, but actual test content does not and is identical to previous test.

Reply:
You might almost think I just copy-pasted without thinking about it...